src/mpc2k.c compilation fails with 'error: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size 18 [-Werror=format-truncation=]'

On Ubuntu 18.10 (gcc 8.2.0), when trying to compile master (8ddc442) using the README.md recommended workflow

```sh
./autogen.sh
./configure --enable-werror
make
```

it fails with

```gcc
[…]
  CC       src/libsndfile_la-mpc2k.lo
src/mpc2k.c: In function ‘mpc2k_write_header.part.0’:
src/mpc2k.c:134:48: error: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size 18 [-Werror=format-truncation=]
  snprintf (sample_name, sizeof (sample_name), "%s                    ", psf->file.name.c) ;
                                                ^~
In file included from /usr/include/stdio.h:873,
                 from src/mpc2k.c:21:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:67:10: note: ‘__builtin___snprintf_chk’ output between 21 and 276 bytes into a destination of size 18
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

It seems that gcc has been getting too picky for that space-fill hack. Do you prefer disabling `-Werror` or would you prefer a pull request that space-fills differently?

--- a/programs/sndfile-deinterleave.c
+++ b/programs/sndfile-deinterleave.c
@@ -116,8 +116,13 @@ main (int argc, char **argv)
 
 	for (ch = 0 ; ch < state.channels ; ch++)
 	{	char filename [520] ;
+		size_t count ;
 
-		snprintf (filename, sizeof (filename), "%s_%02d%s", pathname, ch, ext) ;
+		count = snprintf (filename, sizeof (filename), "%s_%02d%s", pathname, ch, ext) ;
+
+		if (count >= sizeof (filename))
+		{	printf ("File name truncated to %s\n", filename) ;
+			} ;
 
 		if ((state.outfile [ch] = sf_open (filename, SFM_WRITE, &sfinfo)) == NULL)
 		{	printf ("Not able to open output file '%s'\n%s\n", filename, sf_strerror (NULL)) ;

--- a/src/chunk.c
+++ b/src/chunk.c
@@ -219,6 +219,7 @@ psf_save_write_chunk (WRITE_CHUNKS * pchk, const SF_CHUNK_INFO * chunk_info)
 {	union
 	{	uint32_t marker ;
 		char str [5] ;
+		/* Update snprintf() format string below when changing this */
 	} u ;
 	uint32_t len ;
 
@@ -241,7 +242,7 @@ psf_save_write_chunk (WRITE_CHUNKS * pchk, const SF_CHUNK_INFO * chunk_info)
 	len = chunk_info->datalen ;
 	while (len & 3) len ++ ;
 
-	snprintf (u.str, sizeof (u.str), "%s", chunk_info->id) ;
+	snprintf (u.str, sizeof (u.str), "%.4s", chunk_info->id) ;
 
 	pchk->chunks [pchk->used].hash = strlen (chunk_info->id) > 4 ? hash_of_str (chunk_info->id) : u.marker ;
 	pchk->chunks [pchk->used].mark32 = u.marker ;

--- a/src/file_io.c
+++ b/src/file_io.c
@@ -116,33 +116,36 @@ psf_fclose (SF_PRIVATE *psf)
 
 int
 psf_open_rsrc (SF_PRIVATE *psf)
-{
+{	size_t count ;
+
 	if (psf->rsrc.filedes > 0)
 		return 0 ;
 
 	/* Test for MacOSX style resource fork on HPFS or HPFS+ filesystems. */
-	snprintf (psf->rsrc.path.c, sizeof (psf->rsrc.path.c), "%s/..namedfork/rsrc", psf->file.path.c) ;
+	count = snprintf (psf->rsrc.path.c, sizeof (psf->rsrc.path.c), "%s/..namedfork/rsrc", psf->file.path.c) ;
 	psf->error = SFE_NO_ERROR ;
-	if ((psf->rsrc.filedes = psf_open_fd (&psf->rsrc)) >= 0)
-	{	psf->rsrclength = psf_get_filelen_fd (psf->rsrc.filedes) ;
-		if (psf->rsrclength > 0 || (psf->rsrc.mode & SFM_WRITE))
-			return SFE_NO_ERROR ;
-		psf_close_fd (psf->rsrc.filedes) ;
-		psf->rsrc.filedes = -1 ;
-		} ;
+	if (count < sizeof (psf->rsrc.path.c))
+	{	if ((psf->rsrc.filedes = psf_open_fd (&psf->rsrc)) >= 0)
+		{	psf->rsrclength = psf_get_filelen_fd (psf->rsrc.filedes) ;
+			if (psf->rsrclength > 0 || (psf->rsrc.mode & SFM_WRITE))
+				return SFE_NO_ERROR ;
+			psf_close_fd (psf->rsrc.filedes) ;
+			psf->rsrc.filedes = -1 ;
+			} ;
 
-	if (psf->rsrc.filedes == - SFE_BAD_OPEN_MODE)
-	{	psf->error = SFE_BAD_OPEN_MODE ;
-		return psf->error ;
+		if (psf->rsrc.filedes == - SFE_BAD_OPEN_MODE)
+		{	psf->error = SFE_BAD_OPEN_MODE ;
+			return psf->error ;
+			} ;
 		} ;
 
 	/*
 	** Now try for a resource fork stored as a separate file in the same
 	** directory, but preceded with a dot underscore.
 	*/
-	snprintf (psf->rsrc.path.c, sizeof (psf->rsrc.path.c), "%s._%s", psf->file.dir.c, psf->file.name.c) ;
+	count = snprintf (psf->rsrc.path.c, sizeof (psf->rsrc.path.c), "%s._%s", psf->file.dir.c, psf->file.name.c) ;
 	psf->error = SFE_NO_ERROR ;
-	if ((psf->rsrc.filedes = psf_open_fd (&psf->rsrc)) >= 0)
+	if (count < sizeof (psf->rsrc.path.c) && (psf->rsrc.filedes = psf_open_fd (&psf->rsrc)) >= 0)
 	{	psf->rsrclength = psf_get_filelen_fd (psf->rsrc.filedes) ;
 		return SFE_NO_ERROR ;
 		} ;
@@ -151,16 +154,21 @@ psf_open_rsrc (SF_PRIVATE *psf)
 	** Now try for a resource fork stored in a separate file in the
 	** .AppleDouble/ directory.
 	*/
-	snprintf (psf->rsrc.path.c, sizeof (psf->rsrc.path.c), "%s.AppleDouble/%s", psf->file.dir.c, psf->file.name.c) ;
+	count = snprintf (psf->rsrc.path.c, sizeof (psf->rsrc.path.c), "%s.AppleDouble/%s", psf->file.dir.c, psf->file.name.c) ;
 	psf->error = SFE_NO_ERROR ;
-	if ((psf->rsrc.filedes = psf_open_fd (&psf->rsrc)) >= 0)
-	{	psf->rsrclength = psf_get_filelen_fd (psf->rsrc.filedes) ;
-		return SFE_NO_ERROR ;
-		} ;
+	if (count < sizeof (psf->rsrc.path.c))
+	{	if ((psf->rsrc.filedes = psf_open_fd (&psf->rsrc)) >= 0)
+		{	psf->rsrclength = psf_get_filelen_fd (psf->rsrc.filedes) ;
+			return SFE_NO_ERROR ;
+			} ;
 
-	/* No resource file found. */
-	if (psf->rsrc.filedes == -1)
-		psf_log_syserr (psf, errno) ;
+		/* No resource file found. */
+		if (psf->rsrc.filedes == -1)
+			psf_log_syserr (psf, errno) ;
+		}
+	else
+	{	psf->error = SFE_OPEN_FAILED ;
+		} ;
 
 	psf->rsrc.filedes = -1 ;
 

--- a/src/mpc2k.c
+++ b/src/mpc2k.c
@@ -131,7 +131,7 @@ mpc2k_write_header (SF_PRIVATE *psf, int calc_length)
 	if (psf->is_pipe == SF_FALSE)
 		psf_fseek (psf, 0, SEEK_SET) ;
 
-	snprintf (sample_name, sizeof (sample_name), "%s                    ", psf->file.name.c) ;
+	snprintf (sample_name, sizeof (sample_name), "%-*.*s", HEADER_NAME_LEN, HEADER_NAME_LEN, psf->file.name.c) ;
 
 	psf_binheader_writef (psf, "e11b", BHW1 (1), BHW1 (4), BHWv (sample_name), BHWz (HEADER_NAME_LEN)) ;
 	psf_binheader_writef (psf, "e111", BHW1 (100), BHW1 (0), BHW1 ((psf->sf.channels - 1) & 1)) ;

